### PR TITLE
stb: add pkgconfig file

### PR DESCRIPTION
**Pull request by yorickvP**
###### Description of changes

Add a pkg-config file to 'stb' so the include files can be found by it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yorickvP** [comment]
@ofborg build stb curaengine

**Comment by Artturin** [comment]
Does upstream not create one?

**Comment by yorickvP** [comment]
Nope! https://github.com/nothings/stb/issues/1191

File: pkgs/development/libraries/stb/default.nix [hunk 1]
```diff
@@ -14,8 +14,17 @@ stdenv.mkDerivation {
...
 
   installPhase = ''
-    mkdir -p $out/include/stb
+    mkdir -p $out/include/stb $out/lib/pkgconfig
     cp *.h $out/include/stb/
+
+    cat << EOF > $out/lib/pkgconfig/stb.pc
+    prefix=''${out}
+    includedir=\''${prefix}/include/stb
+    Cflags: -I\''${includedir}
```
Comment (alyssais) [inline pkgs/development/libraries/stb/default.nix:23]: The example in this [pkg-config guide](https://people.freedesktop.org/~dbn/pkg-config-guide.html) would have us write:

```
includedir=\''${prefix}/include
Cflags: -I\''${includedir}/stb
```

What do you think?

File: pkgs/development/libraries/stb/default.nix [hunk 2]
```diff
@@ -14,8 +14,17 @@ stdenv.mkDerivation {
...
+    Name: stb
+    Version: ''${version}
```
Comment (alyssais) [inline pkgs/development/libraries/stb/default.nix:25]: Where does this value come from?  Is the escape intentional?

File: pkgs/development/libraries/stb/default.nix [hunk 3]
```diff
@@ -14,8 +14,17 @@ stdenv.mkDerivation {
...
+    Description: stb
```
Comment (alyssais) [inline pkgs/development/libraries/stb/default.nix:26]: Can we have a better description?

**Comment by Artturin** [comment]
created [makePkgconfigItem: init new function to generate pc files](https://github.com/NixOS/nixpkgs/pull/178567)

**Comment by smancill** [comment]
I create the file externally in a library I maintain:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/glm/glm.pc.in

And then use `substituteAll`:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/glm/default.nix#L61=

IMO looks cleaner this way.

**Pull request closed without merging**